contracts/pcv/PCVDepositAggregator.sol

```diff
@@ -0,0 +1,444 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import "./IPCVDepositAggregator.sol";
+import "./PCVDeposit.sol";
+import "../refs/CoreRef.sol";
+import "../external/Decimal.sol";
+import "./balancer/IRewardsAssetManager.sol";
+import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import "../libs/UintArrayOps.sol";
+
+/// @title PCV Deposit Aggregator
+/// @notice A smart contract that aggregates erc20-based PCV deposits and rebalances them according to set weights 
```

Comment:
will probably need an ETH compatible one right? Should still base in WETH but have convertibility of some kind

Reply:
yes, I will make a pr onto this pr that implements this (it will probably just subclass the PCVDepositAggregator, ie an ETHPCVDepositAggregator or such)